File: 70.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        res = [0 for _ in range(n+1)]
        res[0] = 1
        res[1] = 1
        for i in range(2, n+1):
            res[i] = res[i-1] + res[i-2]
        return res[n]

File: 70.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n < 2){
            return 1;
        }
        int first = 1;
        int second = 1;
        for (int i = 2; i<= n;i++){
            int tmp = first + second;
            first = second;
            second = tmp;
        }
        return second;
    }
};
